File: Java/백준/Silver/1764. 듣보잡/듣보잡.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        HashSet<String> heard = new HashSet<>(n * 2);
        for (int i = 0; i < n; i++) {
            heard.add(br.readLine().trim());
        }

        ArrayList<String> ans = new ArrayList<>();
        for (int i = 0; i < m; i++) {
            String name = br.readLine().trim();
            if (heard.contains(name)) ans.add(name);
        }

        Collections.sort(ans);

        StringBuilder sb = new StringBuilder();
        sb.append(ans.size()).append('\n');
        for (String s : ans) sb.append(s).append('\n');
        System.out.print(sb.toString());
    }
}

File: Python/백준/Silver/1764. 듣보잡/듣보잡.py
import sys

def main():
    input = sys.stdin.readline
    n, m = map(int, input().split())

    heard = set()
    for _ in range(n):
        heard.add(input().strip())

    ans = []
    for _ in range(m):
        name = input().strip()
        if name in heard:
            ans.append(name)

    ans.sort()
    print(len(ans))
    print("\n".join(ans))

if __name__ == "__main__":
    main()
